### ISSUE TITLE
Avoid viewport clipping

### ISSUE THREAD

**Issue by emikita** [ocornut/imgui#6393]
Currently all geometry is clipped by rect defined by viewport (defined by ImGui::GetIO().DisplaySize). Is there an easy way how to avoid it? I'm rendering my GUI in VR (by projection ImGui generated geometry from screen space to world space) but my GUI is clipped by the clip rect. Yelow line is left side of the clipping rect:
![image](https://user-images.githubusercontent.com/9430185/235708763-56a7043e-7b70-42d8-aa66-6be660923705.png)

**Comment by ocornut** [comment]
Use an arbitrary large DisplaySize.

**Comment by emikita** [comment]
I did but it affects menu length & maybe some other stuff too, but if there is no other way I can live with that, thanks for quick responce.

**Comment by ocornut** [comment]
Right, it may be worthwhile to introduce per-viewport override for the default clip-rect, which may then be different from viewport’s main rect and work rect. 

Or the override may be for the opposite: we let viewport size be large but we provide a small “prefered” rectangle. We would need to precisely find and list all affected feature to design this carefully on a case by case basis. If you want to investigate this thoroughly to come up with a design we can consider it. 

If the only thing you are seeing is an overly large main menu bar then that’s easy to recreate yourself, see what BeginMainMenuBar() is doing. 

Wonky alternative: I suspect that if you call PushClipRect() manually right after a successful Begin() call, you can override clip rect and get to 95% of what you are looking for….. but title bar contents would still be clipped.

**Comment by emikita** [comment]
Wonky alternative works for rendering but doesn't work for windows interaction - e.g. I can't move window when dragging point is outside of clip rect.
